ansible/roles/test/tasks/snmp/lldp.yml

```diff
@@ -0,0 +1,112 @@
+  and not {{ snmp_lldp[item] | search('No Such Object currently exists') }}" }
+  with_items:
+    - lldpLocManAddrSubtype
+    - lldpLocManAddr
+    - lldpLocManAddrLen
+    - lldpLocManAddrIfSubtype
+    - lldpLocManAddrIfId
+    - lldpLocManAddrOID
+
+# Check if lldpLocPortTable is present
+- fail:
+    msg: "lldpLocPortTable data missing for interface {{ item.key }}"
+  when: "{{ item.value.description | match('^Ethernet') }}
+  and ( {{ item.value['lldpLocPortNum'] is not defined }}
+  or {{ item.value['lldpLocPortIdSubtype'] is not defined }}
+  or {{ item.value['lldpLocPortId'] is not defined }}
+  or {{ item.value['lldpLocPortDesc'] is not defined }} )"
+  with_dict: "{{ snmp_interfaces }}"
+
+# Check if lldpRemTable is present
+
+- set_fact:
+    active_intf: []
+
+- name: find minigraph lldp neighbor
+  set_fact:
+    minigraph_lldp_nei:  "{{ minigraph_lldp_nei|default({}) | combine({ item.key : item.value}) }}"
+  when: "'server' not in item.value['name'] | lower"
+  with_dict: minigraph_neighbors
+
+- name: Create list of ports with lldpRemTable data
+  when: "{{ item.value['lldpRemTimeMark'] is defined }}
+  and {{ item.value['lldpRemLocalPortNum'] is defined }}
+  and {{ item.value['lldpRemIndex'] is defined }}
+  and {{ item.value['lldpRemChassisIdSubtype'] is defined }}
+  and {{ item.value['lldpRemChassisId'] is defined }}
+  and {{ item.value['lldpRemPortIdSubtype'] is defined }}
+  and {{ item.value['lldpRemPortId'] is defined }}
+  and {{ item.value['lldpRemPortDesc'] is defined }}
+  and {{ item.value['lldpRemSysName'] is defined }}
+  and {{ item.value['lldpRemSysDesc'] is defined }}
+  and {{ item.value['lldpRemSysCapSupported'] is defined }}
+  and {{ item.value['lldpRemSysCapEnabled'] is defined }}"
+  set_fact:
+    active_intf: "{{ active_intf + [item] }}"
+  with_dict: "{{ snmp_interfaces }}"
+
+- debug:
+    msg: "Found {{ active_intf | length }} Ifs with lldpRemTable data\n
+    Minigraph contains {{ minigraph_lldp_nei | length }} neighbors"
+
+- name: Verify lldpRemTable is available on most interfaces
+  assert: {that: "{{ minigraph_lldp_nei | length * 0.8 }} > {{ active_intf | length }}" }
+
+# Check if lldpRemManAddrTable is present
+
+- set_fact:
+    active_intf: []
+
+- name: Create list of ports with lldpRemManAddr data
+  when: "{{ item.value['lldpRemManAddrSubtype'] is defined }}
+  and {{ item.value['lldpRemManAddr'] is defined }}
+  and {{ item.value['lldpRemManAddr'] is defined }}
+  and {{ item.value['lldpRemManAddrIfSubtype'] is defined }}
+  and {{ item.value['lldpRemManAddrIfId'] is defined }}
+  and {{ item.value['lldpRemManAddrOID'] is defined }}"
+  set_fact:
+   active_intf: "{{ active_intf + [item] }}"
+  with_dict: "{{ snmp_interfaces }}"
+
+- debug:
+    msg: "Found {{ active_intf | length }} Ifs with lldpRemManAddr data\n
+    Minigraph contains {{ minigraph_lldp_nei | length }} neighbors"
+
+- name: Verify lldpRemManAddr is available on most interfaces
```

Comment:
Should we test lldpRemManAddr is available on all interfaces?

Reply:
Taking into consideration that in lldp test we are satisfied with lldp data awailable on 80% of the interfaces, demanding more of the same data here is questionable.
https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/tasks/lldp.yml#L25
I think the intent was to prevent failures caused by testbed troubles - some VMs being down/not sending lldp...